windows/src/desktop/kmconfig/Keyman.System.KMConfigMain.pas

```diff
@@ -0,0 +1,333 @@
+unit Keyman.System.KMConfigMain;
+
+interface
+
+uses
+  Keyman.System.Settings;
+
+type
+  TKMConfig = class
+  private
+    class var IsAdmin: Boolean;
+    class var Settings: TKeymanSettings;
+    class function Import(const Filename: string): Boolean; static;
+    class function Export(const Filename: string): Boolean; static;
+    class function Reset(const ID: string): Boolean; static;
+    class function SetValue(const ID, Value: string): Boolean; static;
+    class function Show(const P2: string): Boolean; static;
+    class procedure Log(const message: string);
+    class procedure LogError(const message: string);
+    class procedure LogWarning(const message: string);
+  public
+    class constructor ClassCreate;
+    class destructor ClassDestroy;
+    class procedure Run; static;
+  end;
+
+implementation
+
+uses
+  System.SysUtils,
+  System.Win.Registry,
+  Winapi.Windows,
+
+  Keyman.System.SettingsManager,
+  Keyman.System.SettingsManagerFile,
+  RegistryKeys;
+
+type
+  TANSIColor = record
+    Green, Grey, Default, Yellow, Red, White: string;
+  end;
+
+var
+  ANSIColor: TANSIColor;
+
+class constructor TKMConfig.ClassCreate;
+var
+  r: TRegistry;
+begin
+  inherited;
+  r := TRegistry.Create;
+  try
+    r.RootKey := HKEY_LOCAL_MACHINE;
+    IsAdmin := r.OpenKey(SRegKey_KeymanEngine_LM, True);
+  finally
+    r.Free;
+  end;
+
+  Settings := TKeymanSettings.Create;
+end;
+
+class destructor TKMConfig.ClassDestroy;
+begin
+  Settings.Free;
+  inherited;
+end;
+
+class procedure TKMConfig.Run;
+var
+  Command: string;
+  Result: Boolean;
+begin
+  Settings := TKeymanSettings.Create;
+  Command := ParamStr(1).ToLower;
+  if Command = 'import' then Result := Import(ParamStr(2))
+  else if Command = 'export' then Result := Export(ParamStr(2))
+  else if Command = 'reset' then Result := Reset(ParamStr(2))
+  else if Command = 'set' then Result := SetValue(ParamStr(2), ParamStr(3))
+  else if Command = 'show' then Result := Show(ParamStr(2))
+  else
+  begin
+    Result := False;
+    writeln('Usage: kmconfig command [options]');
+    writeln('Commands:');
+    writeln('  import <settings.json>   Imports settings into local machine registry');
+    writeln('  export <settings.json>   Exports settings from local machine registry to file');
+    writeln('  reset [id]               Resets one or all settings to default');
```

Comment:
Instead of an empty string `''` to reset all settings, what about using `-a` like show?